astropy/timeseries/periodograms/lombscargle/implementations/scipy_impl.py

```diff
@@ -56,17 +66,30 @@ def lombscargle_scipy(t, y, frequency, normalization="standard", center_data=Tru
     if center_data:
         y = y - y.mean()
 
+    if SCIPY_LT_1_15:
```

Comment:
I think the fit_mean=True version incompatibility should be checked here:
```python
if SCIPY_LT_1_15:
  if fit_mean:
    raise ValueError(...)
  kwargs = {}
else:
  kwargs = {"floating_mean": fit_mean}
```

Reply:
I usually try to keep `raise` statements for invalid user inputs as close to public API calls as possible to minimize confusion. I'm worried that seeing an internal function raise an exception could be interpreted as a bug. If you're unconvinced I'll go with your suggestion.

Reply:
ping @jakevdp
this looks like the one unresolved point in your review